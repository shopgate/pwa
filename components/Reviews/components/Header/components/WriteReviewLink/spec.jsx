import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import {
  mockedStateWithAll,
  mockedStateProductEmpty,
} from 'Components/Reviews/mock';
import WriteReviewLink from './index';

const mockedStore = configureStore();

/**
 * Creates component with provided store state.
 * @param {Object} mockedState Mocked stage.
 * @return {ReactWrapper}
 */
const createComponent = (mockedState) => {
  const Component = (
    <Provider store={mockedStore(mockedState)}>
      <WriteReviewLink />
    </Provider>
  );
  return mount(Component);
};

describe('<WriteReviewLink>', () => {
  let component = null;

  beforeEach(() => {
    jest.resetModules();
  });

  it('should render when current product is set', () => {
    component = createComponent(mockedStateWithAll);
    expect(component).toMatchSnapshot();
    expect(component.find('span').exists()).toBe(true);
  });

  it('should not render when current product is empty', () => {
    component = createComponent(mockedStateProductEmpty);
    expect(component).toMatchSnapshot();
    expect(component.find('span').exists()).toBe(false);
  });
});
