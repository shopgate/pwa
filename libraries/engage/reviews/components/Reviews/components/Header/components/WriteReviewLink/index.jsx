import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import { ITEM_PATH } from '@shopgate/pwa-common-commerce/product/constants';
import { bin2hex } from '@shopgate/pwa-common/helpers/data';
import ButtonLink from '@shopgate/pwa-ui-shared/ButtonLink';

/**
 * Link to add a review.
 * @returns {JSX|null}
 */
const WriteReviewLink = ({ productId }) => (
  <div data-test-id="writeReview" aria-hidden>
    <ButtonLink href={`${ITEM_PATH}/${bin2hex(productId)}/write_review`} noGap>
      <I18n.Text string="reviews.button_add" />
    </ButtonLink>
  </div>
);

WriteReviewLink.propTypes = {
  productId: PropTypes.string.isRequired,
};

export default WriteReviewLink;
