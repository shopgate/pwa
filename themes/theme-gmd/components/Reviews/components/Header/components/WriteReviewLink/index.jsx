import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import { ITEM_PATH } from '@shopgate/pwa-common-commerce/product/constants';
import { bin2hex } from '@shopgate/pwa-common/helpers/data';
import ButtonLink from '@shopgate/pwa-ui-shared/ButtonLink';
import connect from './connector';

/**
 * Link to add a review.
 * @returns {JSX|null}
 */
const WriteReviewLink = ({ productId }) => {
  if (!productId) {
    return null;
  }

  const publicProductId = bin2hex(productId);

  return (
    <div data-test-id="writeReview">
      <ButtonLink href={`${ITEM_PATH}/${publicProductId}/write_review`} noGap>
        <I18n.Text string="reviews.button_add" />
      </ButtonLink>
    </div>
  );
};

WriteReviewLink.propTypes = {
  productId: PropTypes.string,
};

WriteReviewLink.defaultProps = {
  productId: null,
};

export default connect(WriteReviewLink);
