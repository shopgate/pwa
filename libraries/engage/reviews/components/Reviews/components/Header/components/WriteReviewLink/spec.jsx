import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { mockedStateWithAll } from '@shopgate/pwa-common-commerce/reviews/mock';
import WriteReviewLink from './index';

const mockedStore = configureStore();

/**
 * Creates component with provided store state.
 * @param {Object} mockedState Mocked stage.
 * @return {ReactWrapper}
 */
const createComponent = (mockedState) => {
  const Component = (
    <Provider store={mockedStore(mockedState)}>
      <WriteReviewLink productId="foo" />
    </Provider>
  );
  return mount(Component);
};

describe('<WriteReviewLink>', () => {
  let component = null;

  beforeEach(() => {
    jest.resetModules();
  });

  it('should render when current product is set', () => {
    component = createComponent(mockedStateWithAll);
    expect(component).toMatchSnapshot();
    expect(component.find('span').exists()).toBe(true);
  });
});
