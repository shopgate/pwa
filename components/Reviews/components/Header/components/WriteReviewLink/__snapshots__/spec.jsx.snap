// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WriteReviewLink> should render when current product is set 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WriteReviewLink
    productId="foo"
  >
    <div
      aria-hidden={true}
      data-test-id="writeReview"
    >
      <Connect(ButtonLink)
        href="/item/666f6f/write_review"
        noGap={true}
      >
        <ButtonLink
          className=""
          disabled={false}
          flat={true}
          href="/item/666f6f/write_review"
          navigate={[Function]}
          noGap={true}
        >
          <ActionButton
            className=""
            disabled={false}
            flat={true}
            loading={false}
            noGap={true}
            onClick={[Function]}
            rippleClassName=""
            rippleSize={null}
            testId={null}
            type="secondary"
            wrapContent={true}
          >
            <div
              className="css-1yv2jt6"
              data-test-id={null}
            >
              <RippleButton
                className=""
                disabled={false}
                flat={true}
                onClick={[Function]}
                rippleClassName=""
                rippleSize={null}
                testId="Button"
                type="secondary"
                wrapContent={true}
              >
                <Button
                  className=""
                  disabled={false}
                  flat={true}
                  onClick={[Function]}
                  testId="Button"
                  type="secondary"
                  wrapContent={false}
                >
                  <Button
                    className="css-4imzno "
                    disabled={false}
                    onClick={[Function]}
                    testId="Button"
                  >
                    <button
                      className="css-4imzno  css-1rn65c0"
                      data-test-id="Button"
                      disabled={false}
                      onClick={[Function]}
                    >
                      <Ripple
                        className="css-1fscwua "
                        color="#000"
                        disabled={false}
                        fill={true}
                        onClick={[Function]}
                        onComplete={[Function]}
                        overflow={false}
                        size={null}
                      >
                        <div
                          className="css-1fscwua "
                          data-test-id="Ripple"
                          onClick={[Function]}
                          style={null}
                        >
                          <Translate
                            className={null}
                            params={Object {}}
                            role={null}
                            string="reviews.button_add"
                            transform={null}
                          >
                            <span
                              className={null}
                              role={null}
                            >
                              reviews.button_add
                            </span>
                          </Translate>
                        </div>
                      </Ripple>
                    </button>
                  </Button>
                </Button>
              </RippleButton>
            </div>
          </ActionButton>
        </ButtonLink>
      </Connect(ButtonLink)>
    </div>
  </WriteReviewLink>
</Provider>
`;
